app/controllers/application_controller.rb

```diff
@@ -9,4 +11,24 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  # Class accessor for an instance of RedirectMap
+  def self.redirect_map
+    @redirect_map ||= RedirectMap.instance
+  end
+
+  # Instance accessor for the redirect map. This is used in the :before_action
+  # hook.
+  def redirect_map
+    self.class.redirect_map
+  end
+
+  if Rails.env != 'test'
+    redirect_map.load Rails.root.join('config', 'redirect_map.yml')
+  end
```

Comment:
Do you have any idea how long loading in the hash will actually take? I'm wondering about development environment restarts... Maybe it could be lazy load? Like, if nothing else matches in routes, then fall through to this, and only then load the hash? Or maybe load different versions of the file depending on what your environment is? so `redirect_map.test.yml` could be short.